sky/backends/backend_utils.py

```diff
@@ -481,33 +481,16 @@ def add_cluster(
         host_name = cluster_name
         sky_autogen_comment = ('# Added by sky (use `sky stop/down '
                                f'{cluster_name}` to remove)')
-        overwrite = False
+        fix_stale_config = False
         ip = ips[0]
         if docker_user is not None:
             ip = 'localhost'
 
         config_path = os.path.expanduser(cls.ssh_conf_path)
-        with open(config_path, 'r') as f:
-            config = f.readlines()
-
-        ssh_dir = cls.ssh_cluster_path.format('')
-        os.makedirs(os.path.expanduser(ssh_dir), exist_ok=True)
 
-        # Handle Include on top of Config file
-        include_str = f'Include {cls.ssh_cluster_path.format("*")}'
-        for i, line in enumerate(config):
-            config_str = line.strip()
-            if config_str == include_str:
-                break
-            # Did not find Include string. Insert `Include` lines.
-            if 'Host' in config_str:
-                with open(config_path, 'w') as f:
-                    config.insert(0, f'# Added by sky\n{include_str}\n')
-                    f.write(''.join(config).strip())
-                    f.write('\n' * 2)
-                break
-
-        #For backward compatibility
+        # For backward compatibility: before #2706, we wrote the config of SkyPilot clusters
```

Comment:
It seems we can always call the `remove_cluster` in L573? Why do we need to separate the case for whether the config exist in the original `~/.ssh/config` or not?

Reply:
When we have a cluster name in the config, then the `fix_stale_config` is set to True, and the cluster is removed. I don't think it is always called.